frontend/src/metabase/query_builder/actions.js

```diff
@@ -1271,12 +1256,9 @@ export const loadObjectDetailFKReferences = createThunkAction(
 
       // It's possible that while we were running those queries, the object
       // detail id changed. If so, these fk reference are stale and we shouldn't
-      // put them in state.
+      // put them in state. The detail id is used in the query so we check that.
       const updatedQueryResult = getFirstQueryResult(getState());
-      if (
-        getObjectDetailIdValue(queryResult.data) !==
-        getObjectDetailIdValue(updatedQueryResult.data)
-      ) {
+      if (!_.isEqual(queryResult.json_query, updatedQueryResult.json_query)) {
```

Comment:
This change is not related to the bug. I removed `getObjectDetailIdValue` and updated this to use the query which is simpler anyways.